testng-core/src/main/java/org/testng/SuiteTaskExecutor.java

```diff
@@ -42,29 +39,18 @@ public SuiteTaskExecutor(
   public void execute() {
     String name = "suites-";
     if (RuntimeBehavior.favourCustomThreadPoolExecutor()) {
-      IExecutorFactory execFactory = configuration.getExecutorFactory();
-      ITestNGThreadPoolExecutor executor =
-          execFactory.newSuiteExecutor(
-              name,
-              graph,
-              factory,
-              threadPoolSize,
-              threadPoolSize,
-              Integer.MAX_VALUE,
-              TimeUnit.MILLISECONDS,
-              queue,
-              null);
-      executor.run();
-      service = executor;
+      throw new UnsupportedOperationException("This is NO LONGER Supported in TestNG");
```

Comment:
The configuration validation should be done before and only keep one behavior here.

Reply:
I didn't understand what you meant here. This is basically me removing an earlier deprecated feature of a user being able to enable the usage of GraphThreadPoolExecutor, which is now being removed by this changeset.

Reply:
The configuration validation is not at the good place. It should be done at the begining. Here we expect to have the only possible path.

Reply:
The reason why its being done here is because the configuration value is a JVM argument which cannot be set via the TestNG code. We can only read the value via TestNG but there's no way of setting it from within the codebase. Hence the validation exists at this place. On a side note, I am fine with getting rid of this configuration reading entirely.

Reply:
I suggest to have RuntimeBehavior.validate() and check it before the start

Reply:
I am going to remove this read block itself so it wont be needed any more in this case.